### ISSUE TITLE
Memory leak when opening Inspect tool to focus items

### ISSUE THREAD

**Issue by Marie-Shi** [dotnet/winforms#3016]
* .NET Core Version: 
.Net Core 5.0.100-preview.3.20176.4 from master branch + VS2019 with latest core designer

* Have you experienced this same bug with .NET Framework?: 
Yes

**More Info:**

1. This issue can repro from .Net 4.5 to .Net 4.8.
2. This issue can also repro if we just open inspect tool, but not focusing items with the Inspect tool.
3. This issue cannot repro without Inspect tool opening.

**Steps to reproduce:**
1.	Download Process Explorer from here: https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer.
2.	Extract “WindowsForms_Core_Test.zip” and open it in VS.
[WindowsForms_Core_Test.zip](https://github.com/dotnet/winforms/files/4391849/WindowsForms_Core_Test.zip)
3.	Build and run this project, then run “procexp64.exe” to launch Process Explorer.
4.	Open inspect tool and start to focus items, observe private Bytes changes in Process Explorer.

**Actual behavior:**
The private Bytes keep increasing, the change is a bit large. Please see the following Gif:
![Open_Inspect_Core_process](https://user-images.githubusercontent.com/45864017/77727782-0739e100-7036-11ea-984d-dd607f7bde9e.gif)

**Expected behavior:**
The private Bytes have no much changes, the change is so little. The behavior should be like without Inspect tool opening. Please see the following Gif:
![Not_open_Inspect_Core_process](https://user-images.githubusercontent.com/45864017/77727810-115bdf80-7036-11ea-8f8b-697f3242f526.gif)

**Comment by JeremyKuhne** [comment]
Taking an initial look at this I don't see that we're actually leaking any objects. We do allocate a new `InternalAccessibleObject` (about 1K) for every accessibility interface request, but they are all collected when the GC runs. You can see various things like the `HandleCollector` building it's `IntPtr` cache for GDI handles if you look at allocations early after startup, but after these sorts of caches fill out allocations from highlighting with the inspect tool get completely cleared out when the GC runs.

There might be some opportunity to decrease the size of `InternalAccessibleObject` or to reuse them but I haven't validated that.

**Comment by elachlan** [comment]
@dmitrii-drobotov is this something your team looks at? Has this been possibly fixed by the work you and your team have already done?

**Comment by elachlan** [comment]
Related: #7329

**Comment by dmitrii-drobotov** [comment]
@elachlan yes, this seems related, thanks for bringing it up!

@Marie-Shi / @Olina-Zhang could you please re-test this issue on the current .NET 8 build?

**Comment by Amy-Li03** [comment]
Verified this on .NET latest build: 8.0.100-alpha.1.22612.2, the private bytes are increasing slower than before when opening Inspect tool, but still seem to be faster growth than without Inspect tool. Following is the video:

**With Inspect tool:**
![WithInspect](https://user-images.githubusercontent.com/62929087/207240212-9f608731-aebd-459b-bd73-5fdfd614d3f9.gif)

**Without Inspect tool:**
![WithoutInspect](https://user-images.githubusercontent.com/62929087/207240256-6dc6160f-2d2b-426b-b27f-c172ef98a666.gif)

**Comment by dmitrii-drobotov** [comment]
Having investigated it further, I can confirm that there is no memory leak in managed heap, and the memory that's growing has a limit and doesn't seem to increase infinitely. At some point, while still going through controls with Inspect, memory stops increasing on previous rate and growth becomes as if without Inspect. Screenshot from properties page in Process Explorer, where high I/O usage shows when I was holding Tab key:
![image](https://user-images.githubusercontent.com/102954094/209192883-04154771-10ef-4531-b082-b944592da35a.png)

There are also no significant changes in managed heap size after continuous focus changes:
![Screenshot 2022-12-22 181457](https://user-images.githubusercontent.com/102954094/209190148-a9ded7cf-f350-45cf-96ff-4a97030add90.png)

Considering that there is a limit of memory growth with Inspect, and that the size of managed heap doesn't change, we could close this issue as won't fix.

@Amy-Li03 could you please check on your side that after switching controls with Inspect by holding Tab, after some time memory stops growing as fast and there is a limit?

**Comment by Amy-Li03** [comment]
@dmitrii-drobotov yes, after some time memory stops growing as fast as it did at the beginning.

**Comment by dmitrii-drobotov** [comment]
@Amy-Li03 thanks, closing it then.
